include/macros.h

```diff
@@ -102,7 +102,7 @@
 
 #if OOT_DEBUG
 #define PRINTF osSyncPrintf
-#elif defined(__sgi) /* IDO compiler */
+#elif defined(__sgi) && !defined(__GNUC__) && !defined(PERMUTER) && !defined(M2CTX) /* IDO compiler */
```

Comment:
Are all of these checks necessary or could it be simplified?

We (will) have at least 3 printf-style macros now, maybe we could package this up somehow?

Reply:
The issue I was having is some tools define `__sgi` even when using a modern preprocessor which rightly errors on the IDO version of `PRINTF`. I can think of two ways to simplify/package it up:
- `defined(__sgi) && (__STDC_VERSION__ < 199901L)` detects if modern cpp is pretending to be IDO, but it's not so obvious why it's needed
- I believe `#define USE_IDO_PRINTF defined(__sgi) && !defined(__GNUC__) && !defined(PERMUTER) && !defined(M2CTX)` would work if we intend to use the same check for selection other macros in future

Reply:
I guess I was implicitly asking if you wanted to package it up in this PR. It's totally fine if not, one of the places we would use it (#1678) has not yet been merged

Reply:
I'm happy to implement something in this PR, but I'd like to hear what the preferred solution is (either of the options I mentioned or an alternative)

Reply:
Oh sorry. I do think `__STDC_VERSION__` is pretty weird, and the second method is probably better.

Does it work as `__sgi && !__GNUC__ && !PERMUTER && !M2CTX`? It's shorter but I'm not sure if it's confusing

Reply:
That does work, I don't think it's especially confusing so I've gone with that for now